### PR TITLE
release-23.1: add more observability to restore span processing

### PR DESCRIPTION
**Pull request by dt**
see commits.

This might not merge but is intended to be run, at least in custom build form, in a production cluster so it should be reviewed as if it will merge. If it does merge, it would be forward-ported to master and backported to 23.2, but this isn't the priority right now.

### CODE REVIEW

**Review by benbardin — APPROVED** [review]

File: pkg/ccl/backupccl/restore_data_processor.go [hunk 1]
```diff
@@ -276,6 +276,7 @@ func (rd *restoreDataProcessor) openSSTs(
 		readAsOfIter := storage.NewReadAsOfIterator(iter, rd.spec.RestoreTime)
 
 		cleanup := func() {
+			log.VInfof(ctx, 1, "finished with and closing %d files in span [%s-%s)", len(entry.Files), entry.Span.Key, entry.Span.EndKey)
```
Comment (stevendanna) [inline pkg/ccl/backupccl/restore_data_processor.go:280]: `[%s-%s)"` I nearly want to make "%+v" for a roachpb.Span do this by default as it is often much easier to read than the default string format for spans.

File: pkg/ccl/backupccl/restore_data_processor.go [hunk 2]
```diff
@@ -343,6 +351,12 @@ func (rd *restoreDataProcessor) runRestoreWorkers(
 					return done, nil
 				}
 
+				ctx := logtags.AddTag(ctx, "restore-span", fmt.Sprintf("%s", hex.EncodeToString(entry.Span.Key)))
```
Comment (stevendanna) [inline pkg/ccl/backupccl/restore_data_processor.go:354]: nit:

```suggestion
				ctx := logtags.AddTag(ctx, "restore-span", hex.EncodeToString(entry.Span.Key))
```

**Review by stevendanna — APPROVED** [review]
LGTM.

File: pkg/ccl/backupccl/restore_data_processor.go (hunk 2 above)
Comment (dt) [reply pkg/ccl/backupccl/restore_data_processor.go:354]: On further consideration, I think this is actually very much not allowed w.r.t. security -- the pprof profile endpoints shouldn't be able to exfiltrate user data. I switched this to the span index (progressidx) in the master version of this change, and will backport that one instead I think.

https://github.com/cockroachdb/cockroach/pull/119572

File: pkg/ccl/backupccl/restore_data_processor.go (hunk 1 above)
Comment (dt) [reply pkg/ccl/backupccl/restore_data_processor.go:280]: Yeah, the cognitive overhead I find I encounter trying to read the clever/"concise" default printing is real (and if the key contains dashes or negative signs? game over)

**Comment by dt** [comment]
I'm gonna close this in favor of reviewing https://github.com/cockroachdb/cockroach/pull/119572 and then backporting it, and backporting https://github.com/cockroachdb/cockroach/pull/115517 plus the addition to it in #119571.

**Pull request closed without merging**